### PR TITLE
fix: set folder id on flows imported on startup

### PR DESCRIPTION
**Pull request by jordanrfrazier**
Using `LANGFLOW_LOAD_FLOWS_PATH` to import flows on startup did not set the `folder_id` correctly, causing flows to be orphaned in the frontend, with no way to display them. The `api/v1/flows` still lists them, which is why it originally was considered working. 

This PR adds `folder_id` to both existing orphaned flows and newly created flows. 

Note that the current behavior just uses the default_folder as the folder. Ideally, as per #4010, we'd have the ability to scope flows to folders and users. This PR is the first step so as to unblock users from the basic functionality.

### CODE REVIEW

File: src/backend/base/langflow/initial_setup/setup.py [hunk 1]
```diff
@@ -522,29 +531,46 @@ def load_flows_from_directory():
 
                 existing = find_existing_flow(session, flow_id, flow_endpoint_name)
                 if existing:
+                    print(f"Found existing flow: {existing.name}")
```
Comment (ogabrielluiz) [inline src/backend/base/langflow/initial_setup/setup.py:534]:
```suggestion
                    logger.debug(f"Found existing flow: {existing.name}")
```

File: src/backend/base/langflow/initial_setup/setup.py [hunk 2]
```diff
@@ -522,29 +531,46 @@ def load_flows_from_directory():
...
                     flow["user_id"] = user_id
+                    flow["folder_id"] = folder_id
                     flow = Flow.model_validate(flow, from_attributes=True)
                     flow.updated_at = datetime.now(tz=timezone.utc).astimezone()
                     session.add(flow)
 
 
 def find_existing_flow(session, flow_id, flow_endpoint_name):
     if flow_endpoint_name:
+        print(f"flow_endpoint_name: {flow_endpoint_name}")
```
Comment (ogabrielluiz) [inline src/backend/base/langflow/initial_setup/setup.py:566]:
```suggestion
        logger.debug(f"flow_endpoint_name: {flow_endpoint_name}")
```

File: src/backend/base/langflow/initial_setup/setup.py [hunk 3]
```diff
@@ -522,29 +531,46 @@ def load_flows_from_directory():
...
         stmt = select(Flow).where(Flow.endpoint_name == flow_endpoint_name)
         if existing := session.exec(stmt).first():
+            print(f"Found existing flow by endpoint name: {existing.name}")
```
Comment (ogabrielluiz) [inline src/backend/base/langflow/initial_setup/setup.py:569]:
```suggestion
            logger.debug(f"Found existing flow by endpoint name: {existing.name}")
```

File: src/backend/base/langflow/initial_setup/setup.py [hunk 4]
```diff
@@ -522,29 +531,46 @@ def load_flows_from_directory():
...
             return existing
     stmt = select(Flow).where(Flow.id == flow_id)
     if existing := session.exec(stmt).first():
+        print(f"Found existing flow by id: {flow_id}")
```
Comment (ogabrielluiz) [inline src/backend/base/langflow/initial_setup/setup.py:573]:
```suggestion
        logger.debug(f"Found existing flow by id: {flow_id}")
```

File: src/backend/base/langflow/initial_setup/setup.py [hunk 5]
```diff
@@ -522,29 +531,46 @@ def load_flows_from_directory():
...
                     existing.updated_at = datetime.now(tz=timezone.utc).astimezone()
                     existing.user_id = user_id
+
+                    # Generally, folder_id should not be None, but we must check this due to the previous
+                    # behavior where flows could be added and folder_id was None, orphaning
+                    # them within Langflow.
+                    if existing.folder_id is None:
+                        folder_id = get_default_folder_id(session, user_id)
+                        existing.folder_id = folder_id
+
```
Comment (ogabrielluiz) [inline src/backend/base/langflow/initial_setup/setup.py:549]: What do you think about adding a new folder just for this feature?

**Review by ogabrielluiz — CHANGES_REQUESTED** [review]

File: src/backend/base/langflow/initial_setup/setup.py (hunk 5 above)
Comment (jordanrfrazier) [reply src/backend/base/langflow/initial_setup/setup.py:549]: I'd like to allow a way for users to define the folder (and user) structure. Any thoughts on how best we could do that? 

I've thought of allowing `<load_path>/folders/<folder>/flows.json` :shrug:

Comment (ogabrielluiz) [reply src/backend/base/langflow/initial_setup/setup.py:549]: Maybe add a layer above the flow in the json? We are exploring this option in another feature

**Review by ogabrielluiz — APPROVED** [review]
LGTM

**Pull request merged**